src/framework/dropdown/DropdownGroup.jsx

```diff
@@ -0,0 +1,41 @@
+
+import classNames from 'classnames';
+import React, {
+  PropTypes,
+} from 'react';
+
+import Dropdown from './Dropdown.jsx';
+
+const DropdownGroup = (props) => {
+  const options = props.options.reduce(
+    (accumulator, optionList, currentIndex) => (
+      accumulator.concat(optionList.map((option, index) => (
+        Object.assign({}, option, {
+          classes: classNames(props.optionGroupClasses[currentIndex], {
+            'dropdownGroup--first': (accumulator.length > 0 && index === 0),
+          }),
+        })
+      )))
+    ),
+  []);
+
+  const extendedProps = Object.assign({}, props, {
+    options,
+  });
+
+  return (
+    <Dropdown
+      {...extendedProps}
+    />
+  );
+};
+
+DropdownGroup.propTypes = Object.assign({}, Dropdown.propTypes, {
+  optionGroupClasses: PropTypes.array,
```

Comment:
One more thought: This array is not required and should probably also not, which means `props.optionGroupClasses[currentIndex]` would break. As of our usage right now it doesn't matter because we need it anyway. But else more liberal handling would be nice.